src/window.rs

```diff
@@ -129,30 +138,30 @@ impl WindowWrapper {
     }
 
     pub fn synchronize_settings(&mut self) {
-        let editor_title = { EDITOR.lock().title.clone() };
-        if self.title != editor_title {
-            self.title = editor_title;
-            self.window
-                .set_title(&self.title)
-                .expect("Could not set title");
-        }
-
-        let transparency = { SETTINGS.get::<WindowSettings>().transparency };
-        if let Ok(opacity) = self.window.opacity() {
-            if opacity != transparency {
-                self.window.set_opacity(transparency).ok();
-                self.transparency = transparency;
+        if let Some(mut window) = self.window.take() {
```

Comment:
This pattern seems weird to me. This SO answer suggestions there may be better ways: https://stackoverflow.com/a/27363626

Reply:
I'm not too clear on the suggestion. Would match be better?

Reply:
I may be missing something, but to my eyes this:

```
impl Foo {
    fn send(&mut self) {
        if let Some(x) = &mut self.stream {
            *x += 1;
        }
    }
}
```

Feels better to me.

Reply:
Sorry, this is where my lack of familiarity with rust sort of fails me. It looks better, but it looks like something I've tried and got dinged for, because SDL window doesn't derive Copy. I'll give the pattern a shot.

Reply:
pretty sure this shouldn't require copy... But what do I know

Reply:
I'll give this a try!